### PR TITLE
agent-sample: sample agent

### PR DESCRIPTION
**Pull request by t-bast**
I'm adding a new package in Indigo-JS: agent-sample.
This is a way to leverage the monorepo structure and lerna to be able to easily build docker images with the latest changes (and make sure we keep an up-to-date agent sample).
This is getting me one step closer to an easy local setup of an E2E indigo deployment (after this I'll only need to udate strat with a new template and we should be good to go).

---
This change is [Reviewable](https://reviewable.io/reviews/stratumn/indigo-js/183)

### CODE REVIEW

**Comment by jeremie-stratumn** [comment]
:lgtm: this is a great addition!

---

Reviewed 18 of 18 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-:-L0dh0JPe6H0W5WiX0ki:b-6bje0g)*

**Comment by such** [comment]
Thanks! We might want to add license headers everywhere (I know it hurts).

---

Reviewed 18 of 18 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[packages/agent-sample/package.json, line 2 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0dhx-sZtM8-rUb6uY-:-L0dhx-sZtM8-rUb6uY0:b-l3frhy) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-sample/package.json#L2)):*
> ```JSON
> {
>   "name": "agent-sample",
> ```

Is this going to be published or should we make it private regarding npm so we don't push it by mistake?

---

*[packages/agent-ui/Dockerfile.local, line 9 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0diwPiJXwzUIk2m0QB:-L0diwPiJXwzUIk2m0QC:b-us09zq) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-ui/Dockerfile.local#L9)):*
> ```
> FROM node:8
> 
> RUN npm install -g yarn
> ```

From what I read there: https://hub.docker.com/r/yarnpkg/node-yarn/ `yarn` is preinstalled on the image. This would allow us to stick to using `yarn` for installing `lerna` and `serve`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-:-L0dh8B3OKivQCfGGDiG:b-ufjllp)*

**Comment by t-bast** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[packages/agent-sample/package.json, line 2 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0dhx-sZtM8-rUb6uY-:-L0eVtSnXDTRWpKFlmhr:b-y1ura2) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-sample/package.json#L2)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

Is this going to be published or should we make it private regarding npm so we don't push it by mistake?
</blockquote></details>

I think we can publish it, don't you think?

---

*[packages/agent-ui/Dockerfile.local, line 9 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0diwPiJXwzUIk2m0QB:-L0eVz6J4dlD7VwdCXVB:b-99vdl) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-ui/Dockerfile.local#L9)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

From what I read there: https://hub.docker.com/r/yarnpkg/node-yarn/ `yarn` is preinstalled on the image. This would allow us to stick to using `yarn` for installing `lerna` and `serve`
</blockquote></details>

Cool, I'll test it out

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183)*

**Comment by such** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[packages/agent-sample/package.json, line 2 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0dhx-sZtM8-rUb6uY-:-L0eWlKmGwnfA3mkAWaO:b-s9kvsl) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-sample/package.json#L2)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

I think we can publish it, don't you think?
</blockquote></details>

I don't mind publishing it!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183)*

**Comment by t-bast** [comment]
Review status: 11 of 18 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[packages/agent-sample/package.json, line 2 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0dhx-sZtM8-rUb6uY-:-L0eeK2cwEVc9ukmSs8U:b-wzq709) ([raw file](https://github.com/stratumn/indigo-js/blob/c0fd72ca0bd3185a4115436858d71abf5c1f1a3a/packages/agent-sample/package.json#L2)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

I don't mind publishing it!
</blockquote></details>

Perfect :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183)*

**Comment by such** [comment]
:lgtm:

---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-:-L0eglEkAzPTU0nQvFkx:bnfp4nl)*

File: packages/agent-sample/Dockerfile [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM stratumn/agent-js:dev-alpine-v0.18.3
```
Comment (such) [inline packages/agent-sample/Dockerfile]: is this Dockerfile useful?

**Comment by t-bast** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[packages/agent-sample/Dockerfile, line 1 at r3](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0d4CKnjZ_OAClFXyJ8-r3-1:-L0etaVx5mhR-kc3z7uq:b-ayfykz) ([raw file](https://github.com/stratumn/indigo-js/blob/5560bd815883a429b2aff337c2d81438e3eb9b31/packages/agent-sample/Dockerfile#L1)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

is this Dockerfile useful?
</blockquote></details>

It's just to mimick the one created by a strat generate, it will simply use the latest npm packages (instead of latest code).
I think it makes sense to keep it, but if we find that it's completely useless we'll remove it? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183)*

**Comment by such** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[packages/agent-sample/Dockerfile, line 1 at r3](https://reviewable.io:443/reviews/stratumn/indigo-js/183#-L0d4CKnjZ_OAClFXyJ8-r3-1:-L0evCoi8jgNeLrnKHQT:b-vuxgbb) ([raw file](https://github.com/stratumn/indigo-js/blob/5560bd815883a429b2aff337c2d81438e3eb9b31/packages/agent-sample/Dockerfile#L1)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

It's just to mimick the one created by a strat generate, it will simply use the latest npm packages (instead of latest code).
I think it makes sense to keep it, but if we find that it's completely useless we'll remove it? 
</blockquote></details>

My only concern is that we have to keep the agent version up to date... But I guess that as long as it's documented it can be ok.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/183)*

**Pull request merged**